src/StdCheats.sol

```diff
@@ -463,6 +463,19 @@ abstract contract StdCheatsSafe {
             vm.resumeGasMetering();
         }
     }
+
+    // a cheat for fuzzing addresses that are payable only
+    // see https://github.com/foundry-rs/foundry/issues/3631
+    function assumePayable(address addr) internal virtual {
+        // VM address
+        vm.assume(addr != 0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
+        // Console address
+        vm.assume(addr != 0x000000000000000000636F6e736F6c652e6c6f67);
+        // Create2Deployer
+        vm.assume(addr != 0x4e59b44847b379578588920cA78FbF26c0B4956C);
+        // precompiles are also not payable
+        assumeNoPrecompiles(addr);
+    }
```

Comment:
The VM address, console address, and create2 address all will revert when trying to send ETH to them with `.transfer` or `.call` to it (the first two are because the forge VM won't understand the calldata, but on a live chain they can receive ETH because they have no code), so that seems good. 

But:
1. Some precompiles can receive ETH, e.g. `payable(address(2)).transfer(1 ether)` works while `payable(address(1)).transfer(1 ether)` reverts
2. Maybe precompiles on other EVM chains expect to receive ETH transfers? I'm not sure

So perhaps we remove the `assumeNoPrecompiles`, and users can manually stack the two if they want? Not certain on the best option 🤔

Let's also add a test or two like we have for `assumeNoPrecompiles`